shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -0,0 +1,174 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.h"
+
+#include <AudioToolbox/AudioToolbox.h>
+#include <Foundation/Foundation.h>
+#include <UIKit/UIApplication.h>
+#include <UIKit/UIKit.h>
+
+static NSDictionary* GetDirectoryOfType(NSSearchPathDirectory dir) {
+  NSArray* paths =
+      NSSearchPathForDirectoriesInDomains(dir, NSUserDomainMask, YES);
+  if (paths.count == 0)
+    return nil;
+  return @{ @"path": paths.firstObject };
+}
+
+namespace flutter {
+namespace platform {
+
+// TODO(abarth): Move these definitions from system_chrome_impl.cc to here.
+extern const char* const kOrientationUpdateNotificationName;
+extern const char* const kOrientationUpdateNotificationKey;
+extern const char* const kOverlayStyleUpdateNotificationName;
+extern const char* const kOverlayStyleUpdateNotificationKey;
+
+}  // namespace platform
+}  // namespace flutter
+
+using namespace flutter::platform;
+
+@implementation FlutterPlatformPlugin
+
+@synthesize messageName = _messageName;
+
+- (instancetype) init {
+  self = [super init];
+  if (self)
+    self->_messageName = @"flutter/platform";
```

Comment:
This pattern is a bit dangerous. Since the string is technically `autoreleased`, you should have expected the `_messageName` to be garbage after the current auto release pool activation popped. It just happened to work here because you ended up with a `__NSCFConstantString` (a compile time constant) because you used the object literal syntax. But not all object literal syntax usage gives you compile time constants (dictionaries and arrays for example are still heap allocated).

A more idiomatic use of this would be to `+1` retain the string here and release it in a `dealloc`. Also, the `self->` is redundant.

This is more of a nit, but have run into issues while refactoring and such. Feel free to ignore the comment though. It does work.